File: android/src/main/java/expo/modules/gaodemap/overlays/PolygonViewModule.kt
package expo.modules.gaodemap.overlays

import expo.modules.kotlin.modules.Module
import expo.modules.kotlin.modules.ModuleDefinition

/**
 * Polygon 视图 Module
 */
class PolygonViewModule : Module() {
  override fun definition() = ModuleDefinition {
    Name("PolygonView")

    View(PolygonView::class) {
      Events("onPress")
      
      Prop<List<Map<String, Double>>>("points") { view: PolygonView, points ->
        view.setPoints(points)
      }
      
      Prop<Int>("fillColor") { view: PolygonView, color ->
        view.setFillColor(color)
      }
      
      Prop<Int>("strokeColor") { view: PolygonView, color ->
        view.setStrokeColor(color)
      }
      
      Prop<Float>("strokeWidth") { view: PolygonView, width ->
        view.setStrokeWidth(width)
      }

      Prop<Float>("zIndex") { view: PolygonView, zIndex ->
        view.setZIndex(zIndex)
      }
    }
  }
}

File: ios/overlays/PolygonViewModule.swift
import ExpoModulesCore

public class PolygonViewModule: Module {
    public func definition() -> ModuleDefinition {
        Name("PolygonView")
        
        View(PolygonView.self) {
            Events("onPress")
            
            Prop("points") { (view: PolygonView, points: [[String: Double]]) in
                view.setPoints(points)
            }
            
            Prop("fillColor") { (view: PolygonView, color: String) in
                view.setFillColor(color)
            }
            
            Prop("strokeColor") { (view: PolygonView, color: String) in
                view.setStrokeColor(color)
            }
            
            Prop("strokeWidth") { (view: PolygonView, width: Double) in
                view.setStrokeWidth(Float(width))
            }
        }
    }
}